File: mock-messaging-client/src/main/java/app/ApplicationConfiguration.java
package app;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;
import org.springframework.cloud.stream.annotation.EnableBinding;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.concurrent.Executors;

@Configuration
@EnableBinding(TemperatureSource.class)
public class ApplicationConfiguration {

    @Bean
    public Scheduler scheduler() {
        return Schedulers.from(
            Executors.newSingleThreadExecutor()
        );
    }
}

File: mock-messaging-client/src/main/kotlin/app/ApplicationConfiguration.kt
package app

import org.springframework.amqp.core.AmqpTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.integration.amqp.dsl.Amqp
import org.springframework.integration.channel.DirectChannel
import org.springframework.integration.dsl.IntegrationFlow
import org.springframework.integration.dsl.IntegrationFlows

@Configuration
open class ApplicationConfiguration {

    @Autowired private lateinit var amqpTemplate: AmqpTemplate

    @Bean
    open fun fakeTemperatureChannel() = DirectChannel()

    @Bean
    open fun fakeTemperatureToRabbitFlow(amqpTemplate: AmqpTemplate): IntegrationFlow {
        return IntegrationFlows
            .from(fakeTemperatureChannel())
            .handle(Amqp.outboundAdapter(amqpTemplate).exchangeName("temperature"))
            .get()
    }
}
